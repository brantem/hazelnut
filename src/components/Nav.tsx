import Link from 'next/link';
import { useRouter } from 'next/router';

import { Navigation } from 'types/shared';

type NavItemProps = {
  icon: React.ReactNode;
  href: string;
  children: React.ReactNode;
};

const NavItem = ({ icon, href, children }: NavItemProps) => {
  const router = useRouter();
  const isActive = router.pathname === href;

  return (
    <Link href={href}>
      <a
        className="flex flex-1 items-center justify-center px-4 py-3 text-center font-medium"
        aria-current={isActive ? 'page' : undefined}
      >
        <div className="h-6 w-6">{icon}</div>
        {isActive && <span className="ml-2">{children}</span>}
      </a>
    </Link>
  );
};

const Nav = ({ navigations }: { navigations: Navigation[] }) => {
  return (
    <nav className="flex w-full divide-x divide-neutral-200 border-t border-t-neutral-200">
      {navigations.map((navigation, i) => (
        <NavItem key={i} icon={navigation.icon} href={navigation.href}>
          {navigation.text}
        </NavItem>
      ))}
    </nav>
  );
};

export default Nav;
