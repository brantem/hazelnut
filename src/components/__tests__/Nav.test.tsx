import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import Nav from 'components/Nav';

vi.mock('next/router', () => ({
  useRouter() {
    return {
      pathname: '/',
    };
  },
}));

test('Nav', () => {
  const { container } = render(
    <Nav
      navigations={[
        { icon: <span />, href: '/', text: 'A' },
        { icon: <span />, href: '/b', text: 'B' },
      ]}
    />,
  );

  expect(container).toMatchSnapshot();
  expect(screen.getByText('A').parentNode).toHaveAttribute('aria-current', 'page');
  expect(screen.queryByText('B')).not.toBeInTheDocument();
});
