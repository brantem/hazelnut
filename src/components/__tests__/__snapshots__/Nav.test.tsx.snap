// Vitest Snapshot v1

exports[`Nav 1`] = `
<div>
  <nav
    class="flex w-full divide-x divide-neutral-200 border-t border-t-neutral-200"
  >
    <a
      aria-current="page"
      class="flex flex-1 items-center justify-center px-4 py-3 text-center font-medium"
      href="/"
    >
      <div
        class="h-6 w-6"
      >
        <span />
      </div>
      <span
        class="ml-2"
      >
        A
      </span>
    </a>
    <a
      class="flex flex-1 items-center justify-center px-4 py-3 text-center font-medium"
      href="/b"
    >
      <div
        class="h-6 w-6"
      >
        <span />
      </div>
    </a>
  </nav>
</div>
`;
